rust/gitxetcore/src/git_integration/repo_migration.rs

```diff
@@ -0,0 +1,675 @@
+                if *p_count == 0 {
+                    processing_queue.push(*parent_idx);
+                }
+            }
+
+            // Add in the new tree.
+            tr_map.insert(src_tree_id, new_tree_id);
+
+            progress_reporting.register_progress(Some(1), None);
+        }
+
+        // Now, all the trees should be converted. Make sure this is the case.
+        #[cfg(test)]
+        {
+            for (tr, c) in trees {
+                assert_eq!(c, 0);
+                let Some(new_id) = tr_map.get(&tr.id()) else {
+                    panic!("{:?} not in tr_map.", tr.id());
+                };
+                assert!(dest.find_tree(*new_id).is_ok())
+            }
+        }
+        progress_reporting.finalize();
+    }
+
+    // Now that all the trees have been converted, go through  and convert all the commits.
+    // Again, this has to be done in order, so build the network of parents
+    {
+        let progress_reporting =
+            DataProgressReporter::new("Importing commits", Some(commits_to_convert.len()), None);
+
+        let mut commit_children = HashMap::<Oid, Vec<usize>>::new();
+
+        let mut commits = Vec::<(Commit, usize)>::new();
+
+        let mut processing_queue = Vec::new();
+
+        for c_oid in commits_to_convert.iter() {
+            let commit = src.find_commit(*c_oid)?;
+
+            let c_idx = commits.len();
+
+            for p in commit.parents() {
+                commit_children.entry(p.id()).or_default().push(c_idx);
+            }
+
+            let parent_count = commit.parent_count();
+            commits.push((commit, parent_count));
+
+            if parent_count == 0 {
+                processing_queue.push(c_idx);
+            }
+        }
+
+        while let Some(c_idx) = processing_queue.pop() {
+            let (old_commit_id, new_commit_id) = {
+                // All parents have been converted.
+                assert_eq!(commits[c_idx].1, 0);
+
+                let src_commit = &commits[c_idx].0;
+
+                let mut translated_parent_commits = vec![];
+
+                for parent in src_commit.parents() {
+                    let Some(translated_oid) = tr_map.get(&parent.id()) else {
+                        error!("Repo migration logic error: commit parent not translated.");
+                        continue;
+                    };
+
+                    let new_commit = dest.find_commit(*translated_oid)?;
+
+                    translated_parent_commits.push(new_commit);
+                }
+
+                let Some(new_tree_id) = tr_map.get(&src_commit.tree_id()) else {
+                    panic!("Logic Error: commit tree id not translated.");
+                };
+
+                let new_tree = dest.find_tree(*new_tree_id)?;
+
+                // Create a new commit in the destination repository
+                let new_commit_id = dest.commit(
+                    None, // This updates the HEAD to point to the new commit
```

Comment:
I noticed that API instead says `Use "HEAD" to update the HEAD of the current branch and make it point to this commit`, change the comment if this is desired

Reply:
Feature request: is it possible to keep the original commit timestamp? I would be very happy to see my history actually doesn't change. But this may be a product question

Reply:
The comment here is wrong -- will update.  None does not update HEAD. 

It does, however, make me want to ensure that HEAD is indeed set to main.

Reply:
The timestamp is contained in the committer signature object, which we just copy over, so it should be preserved.